gammapy/maps/tests/test_geom.py

```diff
@@ -225,3 +225,23 @@ def test_mapaxis_repr():
 def test_mapcoord_repr():
     coord = MapCoord({"lon": 0, "lat": 0, "energy": 5})
     assert "MapCoord" in repr(coord)
+
+mapaxis_geoms_node_type_unit = [
```

Comment:
I'm not sure this list of test cases is useful.

Naively I would have thought you put one test case where `==` and one where `!=`.
For me, that would be enough (it would e.g. find the missing `__ne__` for Python 2.
And then the next best thing would be to have one "reference" and a copy, and then vary each data member one at a time, checking that the comparison always becomes `!=`. That would show that your `==` is complete.

Currently I think you could have different axis names, and the axes objects would still be classified as equal?

Reply:
Yes. Is the name equality really important here? For serialization yes, but not for arithmetics, no?

Reply:
Well, there can only be one `__eq__`, and IMO it should include all data members, also `name`.

Reply:
OK done. 
Now we check for `MapAxis.name` (case insensitive for serialization).
Tests are now done both for `__eq__` and for `__ne__` changing one property at a time.